torch/csrc/utils/tensor_new.cpp

```diff
@@ -277,7 +276,11 @@ Tensor internal_new_from_data(
 
   auto sizes = compute_sizes(data);
   ScalarType inferred_scalar_type = type_inference ? infer_scalar_type(data) : scalar_type;
-  auto tensor = autograd::make_variable(at::empty(sizes, at::initialTensorOptions().dtype(inferred_scalar_type).pinned_memory(pin_memory)), /*requires_grad=*/false);
+  // This exists to prevent us from tracing the call to empty().  The actual
+  // autograd code doesn't really matter, because requires_grad is always false
+  // here.
+  at::AutoNonVariableTypeMode guard;
```

Comment:
nit: `guard_` for consistency

Reply:
I probably should change the other guards around. Trailing underscore reserved for properties on C++ classes...